iothub_client/inc/iothub_client_properties.h

```diff
@@ -142,76 +147,76 @@ typedef struct IOTHUB_CLIENT_DESERIALIZED_PROPERTY_TAG {
     /** @brief Number of bytes in valueLength.  Does not include null-terminator if IOTHUB_CLIENT_PROPERTY_VALUE_STRING 
     *          is set */
     size_t valueLength;
-} IOTHUB_CLIENT_DESERIALIZED_PROPERTY;
+} IOTHUB_CLIENT_PROPERTY_PARSED;
```

Comment:
I see 3 types of IOTHUB_CLIENT_PROPERTY_* that are to me confusing:
- `IOTHUB_CLIENT_PROPERTY_PARSED`
- `IOTHUB_CLIENT_PROPERTY_WRITABLE_RESPONSE`
- `IOTHUB_CLIENT_PROPERTY_REPORTED`

The API is not self-describing (i.e. the dev shouldn't need to read the documentation to understand it).
Second, the classification between the 3 is done on what I see as disjunct classes:

- One is parsed - are the others not parsed? Are others constructed by the application - i.e. the Writer APIs? If so, we should probably tag them as "parser" properties vs "writer" properties?
- I expect there is a "Request" vs a "Response" distinction. (Is the "Reported" a R/O Request property?)
- In Twin we use the classification of "Reported" vs "Desired". Could we call the Parsed property a "Desired" property?